types/src/contract_messages/messages.rs

```diff
@@ -238,6 +266,16 @@ impl Message {
     pub fn topic_key(&self) -> Key {
         Key::message_topic(self.entity_addr, self.topic_name_hash)
     }
+
+    /// Returns the checksum of the message.
+    pub fn checksum(&self) -> Result<MessageChecksum, bytesrepr::Error> {
+        let mut payload_bytes = self.message.to_bytes()?;
+        let mut block_index_bytes = self.block_index.to_bytes()?;
+        block_index_bytes.append(&mut payload_bytes);
+
+        let checksum = crypto::blake2b(block_index_bytes);
```

Comment:
```suggestion
        let input = (&self.block_index, &self.message).to_bytes()?;
        let checksum = crypto::blake2b(input);
```

Reply:
out of curiosity, why would you prefer to bytes over a tuple here? iirc we generally do the append style for multi value hashups.

Reply:
Simpler code (at least IMO) but also fewer allocations 1 vs 2 or even 3 (1 per call to `to_bytes()` and possibly one in `append()`).

Reply:
Yup, good suggestion, changed in: https://github.com/casper-network/casper-node/pull/4550/commits/4550ba389d5e0871c090f901c4c5ac5bd29648fe